packages/settingeditor/src/pluginlist.tsx

```diff
@@ -49,178 +89,117 @@ export class PluginList extends Widget {
-
-    if (id === this._selection) {
-      return;
-    }
-
-    if (!id) {
-      while (!id && target !== this.node) {
-        target = target.parentElement as HTMLElement;
-        id = target.getAttribute('data-id');
-      }
-    }
+  private _evtMousedown(event: React.MouseEvent<HTMLButtonElement>): void {
+    const target = event.currentTarget;
+    const id = target.getAttribute('data-id');
 
     if (!id) {
       return;
     }
 
-    this._confirm()
-      .then(() => {
-        this._scrollTop = this.scrollTop;
-        this._selection = id!;
-        this._changed.emit(undefined);
-        this.update();
-      })
-      .catch(() => {
-        /* no op */
-      });
-  }
-
-  protected translator: ITranslator;
-  private _changed = new Signal<this, void>(this);
-  private _confirm: () => Promise<void>;
-  private _scrollTop: number | undefined = 0;
-  private _selection = '';
-}
-
-/**
- * A namespace for `PluginList` statics.
- */
-export namespace PluginList {
-  /**
-   * The instantiation options for a plugin list.
-   */
-  export interface IOptions {
-    /**
-     * A function that allows for asynchronously confirming a selection.
-     *
-     * #### Notest
-     * If the promise returned by the function resolves, then the selection will
-     * succeed and emit an event. If the promise rejects, the selection is not
-     * made.
-     */
-    confirm: () => Promise<void>;
-
-    /**
-     * The setting registry for the plugin list.
-     */
-    registry: ISettingRegistry;
-
-    /**
-     * The setting registry for the plugin list.
-     */
-    translator?: ITranslator;
+    if (this._confirm) {
+      this._confirm(id)
+        .then(() => {
+          this.selection = id!;
+          this._changed.emit(undefined);
+          this.update();
+        })
+        .catch(() => {
+          /* no op */
+        });
+    } else {
+      this._scrollTop = this.scrollTop;
+      this._selection = id!;
+      this._handleSelectSignal.emit(id!);
+      this._changed.emit(undefined);
+      this.update();
+    }
   }
-}
-
-/**
- * A namespace for private module data.
- */
-namespace Private {
-  /**
-   * The JupyterLab plugin schema key for the setting editor
-   * icon class of a plugin.
-   */
-  const ICON_KEY = 'jupyter.lab.setting-icon';
-
-  /**
-   * The JupyterLab plugin schema key for the setting editor
-   * icon class of a plugin.
-   */
-  const ICON_CLASS_KEY = 'jupyter.lab.setting-icon-class';
-
-  /**
-   * The JupyterLab plugin schema key for the setting editor
-   * icon label of a plugin.
-   */
-  const ICON_LABEL_KEY = 'jupyter.lab.setting-icon-label';
 
   /**
    * Check the plugin for a rendering hint's value.
    *
    * #### Notes
    * The order of priority for overridden hints is as follows, from most
    * important to least:
-   * 1. Data set by the end user in a settings file.
+   * import { showDialog } from '@jupyterlab/apputils';
+1. Data set by the end user in a settings file.
```

Comment:
Sporious copy-paste?

```suggestion
   * 1. Data set by the end user in a settings file.
```